developer_docs/RELEASE.md

```diff
@@ -0,0 +1,58 @@
+# Releasing a new version
+
+This guide aims to provide guidance on how to release new versions of `terraform-provider-ec`.
+
+- [Releasing a new version](#releasing-a-new-version)
+  - [Prerequisites](#prerequisites)
+    - [Make sure `VERSION` is up to date](#make-sure-version-is-up-to-date)
+    - [Ensure the `elastic/cloud-sdk-go` dependency is up to date](#ensure-the-elasticcloud-sdk-go-dependency-is-up-to-date)
+    - [Generating a changelog for the new version](#generating-a-changelog-for-the-new-version)
+      - [Patch version changelog](#patch-version-changelog)
+  - [Executing the release](#executing-the-release)
+  - [Post release tasks](#post-release-tasks)
+
+## Prerequisites
+
+Releasing a new version implies that there have been changes in the source code which are meant to be released for wider consumption. Before releasing a new version there's some prerequisites that have to be checked.
+
+### Make sure `VERSION` is up to date
+
+**Since the `VERSION` is now updated via github actions, just double check that it is updated, and if not, manually do so**.
+
+Since the source has changed, we need to update the current committed version to a higher version so that the release is published.
+
+The version is currently defined in the [Makefile](./Makefile) as an exported environment variable called `VERSION` in the [SEMVER](https://semver.org) format: `MAJOR.MINOR.BUG`
```

Comment:
`MAJOR.MINOR.BUG` -> `MAJOR.MINOR.PATCH` bc `If a patch version needs to be released` down below.